.github/workflows/scripts/build.py

```diff
@@ -37,54 +87,82 @@ def setup_llvm(env_out: dict) -> None:
         else:
             out = get_cache_home() / 'llvm15'
             url = 'https://github.com/taichi-dev/taichi_assets/releases/download/llvm15/taichi-llvm-15-linux.zip'
+        download_dep(url, out, strip=1)
     elif (u.system, u.machine) == ('Darwin', 'arm64'):
         out = get_cache_home() / 'llvm15-m1'
         url = 'https://github.com/taichi-dev/taichi_assets/releases/download/llvm15/taichi-llvm-15-m1.zip'
+        download_dep(url, out, strip=1)
     elif (u.system, u.machine) == ('Darwin', 'x86_64'):
         out = get_cache_home() / 'llvm15-mac'
         url = 'https://github.com/taichi-dev/taichi_assets/releases/download/llvm15/llvm-15-mac10.15.zip'
+        download_dep(url, out, strip=1)
+    elif (u.system, u.machine) == ('Windows', 'AMD64'):
+        out = get_cache_home() / 'llvm15'
+        url = 'https://github.com/python3kgae/taichi_assets/releases/download/llvm15_vs2019_clang/taichi-llvm-15.0.0-msvc2019.zip'
+        download_dep(url, out, strip=0)
     else:
         raise RuntimeError(f'Unsupported platform: {u.system} {u.machine}')
 
-    download_dep(url, out, strip=1)
-    env_out['LLVM_DIR'] = str(out)
+    path_prepend('PATH', out / 'bin')
+    os.environ['LLVM_DIR'] = str(out)
 
 
 @banner('Setup Vulkan 1.3.236.0')
-def setup_vulkan(env: dict):
+def setup_vulkan():
     u = platform.uname()
     if u.system == "Linux":
         url = 'https://sdk.lunarg.com/sdk/download/1.3.236.0/linux/vulkansdk-linux-x86_64-1.3.236.0.tar.gz'
         prefix = get_cache_home() / 'vulkan-1.3.236.0'
         download_dep(url, prefix, strip=1)
         sdk = prefix / 'x86_64'
-        env['VULKAN_SDK'] = str(sdk)
-        env['PATH'] = str(sdk / "bin") + ':' + env["PATH"]
-        env['LD_LIBRARY_PATH'] = str(sdk / "lib") + ':' + env.get(
-            "LD_LIBRARY_PATH", "")
-        env['VK_LAYER_PATH'] = str(sdk / 'etc' / 'vulkan' / 'explicit_layer.d')
+        os.environ['VULKAN_SDK'] = str(sdk)
+        path_prepend('PATH', sdk / "bin")
+        path_prepend('LD_LIBRARY_PATH', sdk / 'lib')
+        os.environ['VK_LAYER_PATH'] = str(sdk / 'etc' / 'vulkan' /
+                                          'explicit_layer.d')
     # elif (u.system, u.machine) == ("Darwin", "arm64"):
     # elif (u.system, u.machine) == ("Darwin", "x86_64"):
-    # elif u.system == "Windows":
+    elif (u.system, u.machine) == ('Windows', 'AMD64'):
+        url = 'https://sdk.lunarg.com/sdk/download/1.3.236.0/windows/VulkanSDK-1.3.236.0-Installer.exe'
```

Comment:
IMO, in a one-click build script we shouldn't install a full SDK on the user's machine, it's too intrusive. Building Vulkan backend does not requrie the SDK, the SDK is only needed to load validation layer. We should instruct users install their own SDK (and the windows installer will setup the VULKAN_SDK variable correctly)

Reply:
- For now this script is only targeting CI environment so this is fine.
- I totally agree the 'too intrusive' concern so every dependency is explicitly placed at a cache dir (See `get_cache_home`) including Vulkan SDK.
- When this script evolves into a developer facing script, all these dependency installing will be optional and be documented.